lightning/backend/local.go

```diff
@@ -305,6 +334,26 @@ func NewLocalBackend(
 	return MakeBackend(local), nil
 }
 
+// lockForImport locks the local file for importing.
+func (local *local) lockForImport(engineId uuid.UUID, state importMutexState) bool {
+	if e, ok := local.engines.Load(engineId); ok {
+		engine := e.(*LocalFile)
+		engine.lockForImport(state)
```

Comment:
@lance6716 we don't need UUID-level lock?

Reply:
#493 will reopen this engineFile. So we need to lock it when resetengine.

Reply:
Need. I just think this PR should be merged into master, and #493 will adjust its content later 😂

Reply:
i'm confused in https://github.com/pingcap/tidb-lightning/pull/543#pullrequestreview-561469974

Reply:
@Little-Wallace maybe you could explain a bit 😅

Reply:
Emmmmm.  #493  will open a new db instance and update it to `localFile`. But this db instance may be called in another goroutine, which is `writeRowsLoop` and try to ingest a sst file into itself. So we must add a lock to prevent data race.

Reply:
if #493 will split `ResetEngine` and `CleanupEngine` to two behaviour, which the former didn't remove `LocalFile` from `local.engines`, I think we don't need UUID-level lock then 😂 because `LocalFile` is not changed after bounded to UUID in map

Reply:
Oh. Yes. We only need to lock the engineFile rather than uuid-level lock. I think the mutex in engineFile is easy to understand

Reply:
PTAL

Reply:
I think it's ok to only depend on the LocalFile level lock after changing the reset engine operation inplace.